ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs

```diff
@@ -1,22 +1,95 @@
 {-# LANGUAGE AllowAmbiguousTypes   #-}
 {-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeOperators         #-}
 
 module Ouroboros.Consensus.Protocol.Translate (TranslateProto (..)) where
 
-import           Ouroboros.Consensus.Protocol.Abstract
+import           Cardano.Crypto.DSIGN (VerKeyDSIGN)
+import           Cardano.Crypto.VRF (VerKeyVRF)
+import qualified Cardano.Ledger.Chain as SL
+import           Cardano.Ledger.Crypto (ADDRHASH, HASH)
+import qualified Cardano.Ledger.PoolDistr as SL
+import qualified Cardano.Protocol.TPraos.API as SL
+import qualified Cardano.Protocol.TPraos.Rules.Prtcl as SL
+import qualified Cardano.Protocol.TPraos.Rules.Tickn as SL
+import           Data.Coerce (coerce)
+import qualified Data.Map.Strict as Map
+import qualified Ouroboros.Consensus.Protocol.Abstract as Abstract
+import           Ouroboros.Consensus.Protocol.Praos (Praos, PraosState (..))
+import           Ouroboros.Consensus.Protocol.Praos.Views
+                     (LedgerView (lvMaxBodySize, lvMaxHeaderSize, lvProtocolVersion))
+import qualified Ouroboros.Consensus.Protocol.Praos.Views as Views
+import           Ouroboros.Consensus.Protocol.TPraos (TPraos,
+                     TPraosState (tpraosStateChainDepState, tpraosStateLastSlot))
 
 -- | Translate across protocols
 class TranslateProto protoFrom protoTo
```

Comment:
Maybe we could even move this class to `oc-cardano`?

Reply:
I moved it to `Ouroboros.Consensus.Protocol.Abstract` :thinking: